modules/c/examples/code_snippets/main.c

```diff
@@ -1078,3 +1197,392 @@ Placeholder for Date accessors.
+                                                        thisRepl,
+                                                        SimpleReplicationDocumentListener,
+                                                        context);
+
+    // end::add-document-replication-listener[]
+    // tag::remove-document-replication-listener[]
+    // Purpose -- illustrate removal of a listener
+
+    CBLListener_Remove(token_ReplDocListener);
+
+    CBLListener_Remove(token_ReplChangeListener);
+
+    // end::remove-document-replication-listener[]
+
+    // tag::p2p-act-rep-status[]
+    // Purpose -- illustrate use of CBLReplicator_Status()
+
+    CBLReplicatorStatus thisState = CBLReplicator_Status(thisRepl);
+
+    if(thisState.activity==kCBLReplicatorStopped) {
+        if(thisState.error.code==0) {
+            CBLReplicator_Start(thisRepl,false);
+        } else {
+            printf("Replicator stopped -- code %d", thisState.error.code);
+            // ... handle error ...
+        }
+    }
+
+    // end::p2p-act-rep-status[]
+
+
+    // tag::replication-pendingdocuments[]
+
+    FLDict thisPendingIdList =
+        CBLReplicator_PendingDocumentIDs(thisRepl, &err); // <.>
+
+    if(!FLDict_IsEmpty(thisPendingIdList)) {
+        FLDictIterator item;
+        FLDictIterator_Begin(thisPendingIdList, &item);
+        FLValue itemValue;
+        FLString pendingId;
+
+        while(NULL != (FLDictIterator_GetValue(&item))) {
+            pendingId = FLValue_AsString(itemValue);
+
+            if(CBLReplicator_IsDocumentPending(thisRepl,
+                                               pendingId,
+                                               &err)) {
+                // ... process the still pending docid as required <.>
+
+            } else {
+                // Doc Id no longer pending
+                if(err.code==0) {
+                    // No fail so must have already been pushed
+                    printf("Document already pushed");
+                } else {
+                    // Error detected so handle it
+                    printf("Error code %d checking for pendingId", err.code);
+                    break;
+                }
+            }
+            FLDictIterator_Next(&item);
+        }
+
+        FLDictIterator_End(&item);
+        FLValue_Release(itemValue);
+
+    } else {
+        printf("No Pending Id Docs to process");
+    }
+    // end::replication-pendingdocuments[]
+
+
+
+}
+
+// END replication.html >> Monitor section
+
+// BEGIN replication.html >> Stop section
+// PAGE=Data Sync >> Stop
+// URL=https://docs.couchbase.com/couchbase-lite/current/c/replication.html#lbl-repl-stop
+
+static void docs_act_replication_Stop(
+                                       void* context,
+                                       CBLReplicator* argRepl) {
+    // tag::p2p-act-rep-stop[]
+    // Purpose -- show how to stop a replication
+
+    if(CBLReplicator_Status(argRepl).activity!=kCBLReplicatorStopped) {
```

Comment:
Spacing please....

Reply:
Ok. As regards tags, it can cause a problem if there is no empty line prior to the 'end::' line. So I tend to not have one after the tag:: but include one prior to the end::

I'm sensing you prefer fewer blank lines ... so will try to minimize

Reply:
I do prefer fewer blank lines, but what I meant was that there is no space around the `!=` operator.